src/controllers/confirm.company.controller.ts

```diff
@@ -17,10 +18,36 @@ export const get = async (req: Request, res: Response, next: NextFunction) => {
 };
 
 export const post = async (req: Request, res: Response, next: NextFunction) => {
+  try {
+    const company: CompanyProfile = await getCompanyProfile(req.query.companyNumber as string);
+    if (isStatusEligible(company)) {
+      create(req);
+      const url = CONFIRMATION_STATEMENT_TRADING_STATUS.replace(":companyNumber", company.companyNumber);
```

Comment:
An enum for url params has been added since you wrote this, you can replace :companyNumber with urlParams.PARAM_COMPANY_NUMBER from page.urls.ts